### PR TITLE
HAI-2626 Support hidden registry key for billing contact

### PR DESCRIPTION
**Pull request by umeetiusbaar**
# Description

Kaivuilmoitus billing contact has the same registry key rules as the customer ("Työstä vastaava").

## Type of change

- [ ] Bug fix
- [x] New feature
- [ ] Other

# Instructions for testing

1. Open kaivuilmoitus and go to contacts form page
2. Billing contact registry key should behave the same as the customer registry key.

# Checklist:

- [x] I have written new tests (if applicable)
- [x] I have ran the tests myself (if applicable)
- [ ] I have made necessary changes to the documentation, link to confluence
      or other location:

# Other relevant info

Please describe here if there is e.g. some requirements for this change or
other info that the tester/user needs to know.

### CODE REVIEW

File: src/domain/kaivuilmoitus/Contacts.tsx [hunk 1]
```diff
@@ -86,10 +149,15 @@ export default function Contacts() {
           />
           <TextInput
             name="applicationData.invoicingCustomer.registryKey"
-            label={t('form:yhteystiedot:labels:yTunnusTaiHetu')}
+            label={getInvoicingRegistryKeyLabel(t, selectedContactType)}
             required
             autoComplete="on"
             defaultValue={null}
+            helperText={
+              isRegistryKeyHiddenNotificationShown
```
Comment (markohaarni) [inline src/domain/kaivuilmoitus/Contacts.tsx:157]: Displaying helper text could be determined straight from `registryKeyHidden` without the need for useState and useEffect for that.

File: src/domain/kaivuilmoitus/Contacts.tsx [hunk 2]
```diff
@@ -28,12 +52,51 @@ export default function Contacts() {
   const postalAddressRequired = !ovt || !invoicingOperator;
   const ovtRequired = !ovtDisabled && (!streetName || !postalCode || !city);
 
+  const isMountedForSelectedContactType = useRef(false);
+  const isMountedForRegistryKey = useRef(false);
```
Comment (markohaarni) [inline src/domain/kaivuilmoitus/Contacts.tsx:56]: Are both of these isMounted variables needed? Are there some problems if using only one in both useEffects?

File: src/domain/kaivuilmoitus/Contacts.tsx (hunk 1 above)
Comment (umeetiusbaar) [reply src/domain/kaivuilmoitus/Contacts.tsx]: Thank you. Changed it.

File: src/domain/kaivuilmoitus/Contacts.tsx (hunk 2 above)
Comment (umeetiusbaar) [reply src/domain/kaivuilmoitus/Contacts.tsx]: I use the same technique as in ApplicationContacts: a single isMounted that is set in the last useEffect. Also added the same implementation for reverting back to the original hidden value.

**Review by markohaarni — APPROVED** [review]

**Pull request merged**